client/src/graphics/animated_sprite.lua

```diff
@@ -90,8 +83,10 @@ function AnimatedSprite:update()
     self.finished = false
   end
   if (self.playing and not self.finished) then
-    if (self.frameTime <= anim.frames[self.currentFrame][2]) then
-      self.frameTime = self.frameTime + 1/anim.frameDuration
+    if (self.frameTime <= anim.frames[self.currentFrame].duration) then
+      -- this was 1/frameDuration before with frameDuration fixed to 2 (never set in the example script)
+      -- I think running animations at half the speed of what is specified as durations in the script is a bit confusing because it is not explicit
+      self.frameTime = self.frameTime + 0.5
```

Comment:
Bit weird one here, didn't do an entirely decisive change here.
In general I agree with enabling a short hand syntax in the script for animations that use the same duration throughout but it seems to be in conflict with specifying one in addFrame as they are different durations.
We might want to add a separate function that creates an Animation with constant durations per frame that internally calls `addFrame` accordingly so that there is only one frame duration. Makes it less confusing.

Reply:
Yeah, just going up by one can work. Keeping it an integer would help keep it simple. Also, just removing the equal from `self.frameTime <= anim.frames[self.currentFrame].duration` should solve the extra frame too, shouldn't it?

Reply:
Yeah, it would, I think it's a bit easier on the eyes to have a true counter to see on which frame you are for debugging purposes.
So that if `frameCounter` is 1 then you know you're on the first frame of the current frame rather than having to offset in your head.
Just pushed the changes!

Reply:
Ah ok, can see how that makes sense. I'm just used to 0-index coding so it felt right to me, but It's fine.